tests/scr/testSysbox

```diff
@@ -52,6 +126,61 @@ else
 
 	printf "\nSysbox health checking ...\n"
 	bats --tap tests/health/sysbox-health.bats
-fi
 
-exit 0
+	printf "\nRe-enabling Sysbox idmapped-mounts.\n"
+	set_sysbox_idmapped_mount "enable"
+
+	return 0
+}
+
+function run_docker_userns_remap_tests() {
+	printf "\nExecuting Sysbox tests (docker-userns-remap)\n"
+
+	printf "\nConfiguring Docker with userns-remap\n"
+	docker system prune -a -f
+	docker-cfg --userns-remap=enable
+
+	printf "\nExecuting sysbox-mgr tests ... \n"
+	bats --tap tests/sysmgr
+	printf "\nExecuting sysbox-fs tests ... \n"
+	bats --tap tests/sysfs
+	printf "\nExecuting docker tests ... \n"
+	bats --tap tests/docker
+	printf "\nExecuting dind tests ... \n"
+	bats --tap tests/dind
+	printf "\nExecuting cind tests ... \n"
+	bats --tap tests/cind
+
+	# Launch basic kind-flannel test
+	docker system prune -a -f
+	printf "\nExecuting kind testcases with flannel cni ... \n"
+	bats --tap tests/kind/kind-flannel.bats
+	docker system prune -a -f
+
+	# Launch sysbox-pod test-suite.
+	./tests/pods/pods.sh
+
+	printf "\nSysbox health checking ...\n"
+	bats --tap tests/health/sysbox-health.bats
+
+	# TODO: disable all uid shifting in sysbox and verify things still work
+
+	printf "\nReverting Docker userns-remap config\n"
+	docker system prune -a -f
+	docker-cfg --userns-remap=disable
+}
+
+function main() {
+
+	# argument testName is optional; if present, only that test is executed.
+	if [ $# -eq 1 ]; then
+		printf "\nExecuting $1 ... \n"
+		bats --tap $1
+	else
+		run_idmapped_mount_tests
```

Comment:
It takes long; for a full run, it's ~2 hours. For the CI run, it's about 1.5 hours. This assumes a host that supports idmapped mounts and shiftfs. If the host only supports shiftfs, then it goes 3x faster (i.e., no need to test idmapped + shiftfs and idmapped-only scenarios).

We need to find a way to run these scenarios in parallel. One idea I have is to run each scenario in separate test containers.  But it would require some more re-work of the test framework.

Let me look into this after this PR, given that this PR already includes a many commits.